glslang/MachineIndependent/Scan.h

```diff
@@ -141,23 +148,42 @@ class TInputScanner {
             } else
                 currentChar = lengths[currentSource] - 1;
         }
-        if (peek() == '\n')
+        if (peek() == '\n') {
             --loc[currentSource].line;
+            --logicalSourceLoc.line;
+        }
     }
 
     // for #line override
-    void setLine(int newLine) { loc[getLastValidSourceIndex()].line = newLine; }
-    void setFile(const char* filename) { loc[getLastValidSourceIndex()].name = filename; }
+    void setLine(int newLine)
+    {
+        logicalSourceLoc.line = loc[getLastValidSourceIndex()].line = newLine;
+    }
+
```

Comment:
This method too is used for the #line override.  Now that we've expanded the number of lines of code here it might be worth mentioning the use again.